libraries/lib-wave-track/WaveTrack.cpp

```diff
@@ -358,6 +358,19 @@ void WaveTrack::DoOnProjectTempoChange(
          clip->OnProjectTempoChange(oldTempo, newTempo);
 }
 
+Finally<std::function<void()>>
+WaveTrack::ProvisionallySynchronizeProjectTempo(WaveTrack& other) const
+{
+   const auto originalTempo = other.GetProjectTempo();
+   const auto provisionalTempo = GetProjectTempo();
+   if (!provisionalTempo.has_value() || !originalTempo.has_value())
+      THROW_INCONSISTENCY_EXCEPTION;
```

Comment:
Why this exception?

Reply:
I wonder if instead this should be an assertion that is stated as a precondition.  But then the precondition may need to be repeated for the functions that call this, and so on up the stack until somewhere you can give some convincing argument that the condition is guaranteed.

Reply:
Now I think this throw behaviour is too strict. This method is used so far in `WaveTrack::PasteWaveTrack` and `WaveTrack::ClearAndPaste`. I cannot think of a reason for either of them not to be callable before they are made part of a project, but maybe I'm not thinking hard enough.
A more pragmatic approach would be for this method to assert and return without doing anything instead of throwing, without stating a pre-condition. The assert might reveal a concrete example where this doesn't hold in the future, yet there might not be any consequence at all (e.g. if no stretching occurred), in which case bothering the user with it would just be annoying to everyone.
Ok for assert without pre-condition ?

Reply:
Now I think, let the exception stand.  Rewrite the function as I suggested so it doesn't mutate `other` but constructs a copy.

If the preconditions are not satisfied, throw.  The two places where this function is called are in pasting and I already made sure years ago with exhaustive examination of call paths that these are safe places to throw and have `AudacityApp::OnExceptionInMainLoop` catch it and deal with it.